### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by 0xxlegolas**

### CODE REVIEW

File: test/wklay.test.js [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    it("transfers klay", async () => {
+      const balanceBefore = await wklay.balanceOf(user2);
+      await wklay.transfer(user2, 1, { from: user1 });
+      const balanceAfter = await wklay.balanceOf(user2);
+      balanceAfter
+        .toString()
+        .should.equal(balanceBefore.add(new BN("1")).toString());
+    });
+
+    it("withdraws klay by transferring to address(0)", async () => {
```
Comment (ian0371) [inline test/wklay.test.js:79]: Does this result in withdrawal?

File: contracts/WKLAY.sol [hunk 2]
```diff
@@ -39,9 +40,10 @@ contract WKLAY {
     }
 
     function withdraw(uint256 wad) public {
-        require(balanceOf[msg.sender] >= wad);
+        require(balanceOf[msg.sender] >= wad, "WKLAY: amount exceeds balance");
         balanceOf[msg.sender] -= wad;
-        msg.sender.transfer(wad);
+        (bool success, ) = msg.sender.call{value: wad}("");
```
Comment (ian0371) [inline contracts/WKLAY.sol:45]: now that we're using call, should we test if reentrancy attack is successfully mitigated?

File: test/wklay.test.js (hunk 1 above)
Comment (0xxlegolas) [reply test/wklay.test.js:79]: Yes, using transfer method it is sent to address(0)

File: contracts/WKLAY.sol (hunk 2 above)
Comment (0xxlegolas) [reply contracts/WKLAY.sol:45]: Yes, one of the way is to change the state before we transfer. the state change for balance `balanceOf[msg.sender] -= wad;` is done before transfer. 
We can also inherit reentrancy guard contract

**Comment by 0xxlegolas** [comment]
@ian0371 
For this, the main branch is the standard code which is similar to [WETH](https://github.com/gnosis/canonical-weth/blob/master/contracts/WETH9.sol) and still an active contract, integrated in Uniswap V2/V3, Sushiswap, MEW, etc. So dev branch has the upgraded solidity version. But this can’t be released to the public before we audit.  However the main branch code is audited and being used.

File: test/wklay.test.js (hunk 1 above)
Comment (ian0371) [reply test/wklay.test.js:79]: Oh, I thought sending to address(0) is equivalent to calling `withdraw()`.
Now I see it means the deduction of the sender's balance.

**Comment by ian0371** [comment]
@toniya-klaytn 
I thought approving this means it's ready to be used for production without further audit.
LGTM
While we're upgrading, why not upgrade to the latest solc (v0.8.17)?

**Comment by 0xxlegolas** [comment]
@ian0371 Sure we can upgrade to latest solc version. To use in production, not sure whether we should stick to the WETH standard or use this version after merging. Whats your thought on this ?

**Comment by ian0371** [comment]
@toniya-klaytn Unless we want more features, which I couldn't think of right now, we can stick to the WETH standard.

**Comment by 0xxlegolas** [comment]
Agreed !

**Pull request closed without merging**